CMakeLists.txt

```diff
@@ -144,7 +144,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -z noexecstack")
 endif()
 
 add_library(osal ${SRC_OSAL})
-
+target_link_libraries(osal m)
```

Comment:
can we make a check around that like 

```
if (TARGET m)
  target_link_libraries(osal m)
endif()
```

to only link libm on platforms where it exists?

Reply:
This makes sense.
POSIX requires libm
WIN32 and MINGW automatically link against MSVCRT.DLL [see](https://stackoverflow.com/questions/48134247/why-i-dont-need-the-flag-lm-in-mingw-but-in-linux-i-need-that-explicitly).
For PikeOS and VxWorks I could not find any information yet. Do you know this?

Reply:
While testing the libosal under Windows with WIN32, I saw that not everything defined in the header is implemented, isn't it?
Building tools/logger/main.c under Windows resulted in some linker errors.

Reply:
> This makes sense. POSIX requires libm WIN32 and MINGW automatically link against MSVCRT.DLL [see](https://stackoverflow.com/questions/48134247/why-i-dont-need-the-flag-lm-in-mingw-but-in-linux-i-need-that-explicitly). For PikeOS and VxWorks I could not find any information yet. Do you know this?

checked that. neither vxworks nor pikeos has a libm.

Reply:
> While testing the libosal under Windows with WIN32, I saw that not everything defined in the header is implemented, isn't it? Building tools/logger/main.c under Windows resulted in some linker errors.

yes, as i'm mainly developing for linux (/vxworks/pikeos), the windows part might be a little bit neglected.